### ISSUE TITLE
Error when referencing theme values via useThemeUI after upgrading to @jsxImportSource

### ISSUE THREAD

**Issue by jeffscottward** [system-ui/theme-ui#1777]
**Describe the bug**
After learning I needed to upgrade to the alpha version for `/** @jsxImportSource theme-ui */` to work and doing that
https://github.com/system-ui/theme-ui/issues/1419#issuecomment-754561329, any references (when compiling) to the theme object come back undefined in my Next.js project. It does console.log `theme` correctly if I remove those references though ( but only in a normal component body, not in a `<Global styles ={(theme) =>` expression ). Also it knocks out custom SX theme-ing to override third-party React components.

All this is to say I think that this "solution" for `jsxImportSource` doesn't actually work.

**To Reproduce**
Use `/** @jsxImportSource theme-ui */` in a Next.js project with the correct theme-ui alpha version described above and attempt to reference a value on the theme object via useThemeUI hook.

**Expected behavior**
To compile those value refrences, and not break style overrides for third-party components.

**Screenshots**
<img width="992" alt="Screen Shot 2021-05-26 at 10 58 29 AM" src="https://user-images.githubusercontent.com/1263486/119685598-803bf080-be13-11eb-95fd-12e4d462dcf3.png">
<img width="356" alt="Screen Shot 2021-05-26 at 11 14 29 AM" src="https://user-images.githubusercontent.com/1263486/119685680-92b62a00-be13-11eb-958b-10df84821aca.png">
(background should be blue, text should be teal, third party react-dropdown-select should have correct styles, not box)
<img width="577" alt="Screen Shot 2021-05-26 at 11 17 16 AM" src="https://user-images.githubusercontent.com/1263486/119686109-fdfffc00-be13-11eb-801a-f4d111744be3.png">

**Additional context**
When you try to feed an actual color value in after page load via a hot-reload, it doesn't seem to get picked up.
<img width="341" alt="Screen Shot 2021-05-26 at 11 19 11 AM" src="https://user-images.githubusercontent.com/1263486/119686440-546d3a80-be14-11eb-8a64-769b09c9b14d.png">

**Comment by jeffscottward** [comment]
I just published an isolated && stripped down version of my project to isolate the problem.
https://github.com/jeffscottward/theme-ui-bug-test

It is correctly logging inside Global now but the other problem is the same.

**Comment by hasparus** [comment]
> After learning I needed to upgrade to the alpha version for /** @jsxImportSource theme-ui */ to work

Wait, just so we're clear here. That comment was pretty old. I released 0.9.1 two days ago, and even if you want to use 0.6, you can use stable 0.6, not alpha.

Does this problem occur in a stable version?

**Comment by jeffscottward** [comment]
@hasparus  Yes its the same - just tried it.
v0.9.1 and globally changed to `/** @jsxImportSource theme-ui */`

Just upgraded the test repo too I linked with latest theme-ui and its happening there too. 
Clone it and take a look please? 🙏 Its a tiny test case that's stripped down so someone can diagnose quickly. 

UPDATE: I think i missed this:
The switch from `import { Global } from '@emotion/core'` to `import { Global } from '@emotion/react`
Also - removing any `import { jsx } from 'theme-ui`

TLDR: code that needed updating - not just deps

> > After learning I needed to upgrade to the alpha version for /** @jsxImportSource theme-ui */ to work
> 
> Wait, just so we're clear here. That comment was pretty old. I released 0.9.1 two days ago, and even if you want to use 0.6, you can use stable 0.6, not alpha.
> 
> Does this problem occur in a stable version?
